### ISSUE TITLE
aws_dms_endpoint - Add support for DMS Endpoint settings

### ISSUE THREAD

**Issue by david-fuller988** [hashicorp/terraform-provider-aws#23225]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS plan to deprecate endpoint Extra Connection Attributes and replace with Endpoint settings - there’s no date for this, just a warning in the console.  These are more flexible and combine the existing s3 settings and extra connection attributes, so just one place to set up the configuration.

Without this functionality in Terraform it will not be possible to manage AWS DMS.

### New or Affected Resource(s)

aws_dms_endpoint

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by YakDriver** [comment]
Related #20913

**Comment by stewberg** [comment]
Apologies if not directly relevant, but I found this information helpful until this is moved forward. I've used this interim solution **within my endpoint resource** to add things directly into the **endpoint settings** via terraform apply (since ECAs don't work for my DDL setting - DMS v3.4.7). Example source postgres endpoint: 

  provisioner "local-exec" {
    command = "aws dms modify-endpoint --endpoint-arn '${aws_dms_endpoint.<endpoint_name>.endpoint_arn}' --postgre-sql-settings 'DdlArtifactsSchema=dms_replication'"
  }

**Comment by YakDriver** [comment]
This is resolved for S3 in #28130. The other endpoint types do not have resources yet.
